packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs

```diff
@@ -87,6 +89,39 @@ impl<'c, 'p: 'c, R: PostgresRecord> SelectCompiler<'c, 'p, R> {
         default
     }
 
+    fn pin_entity_table(&mut self, alias: Alias) {
+        let table = Table::Entities.aliased(alias);
+        let temporal_table_info = self.artifacts.temporal_tables.get_or_insert_with(|| {
+            self.artifacts.parameters.push(match self.time_projection {
+                TimeProjection::DecisionTime(projection) => &projection.kernel.timestamp,
+                TimeProjection::TransactionTime(projection) => &projection.kernel.timestamp,
+            });
+
+            TemporalTableInfo {
+                tables: HashSet::new(),
+                parameter_index: self.artifacts.parameters.len(),
+            }
+        });
+
+        if !temporal_table_info.tables.contains(&table) {
+            // Adds the kernel timestamp condition, so for the projected decision time, we use the
+            // transaction time and vice versa.
+            self.statement
+                .where_expression
+                .add_condition(Condition::TimerangeContainsTimestamp(
```

Comment:
This might have been in another PR but can we change this to `TimeRange` not `Timerange` please? (Can be done in follow-up).

"timespan" is one word, but "timerange" is not.

Reply:
Addressed in 070bd2846de71d66d2cbefe04e7cf999e6478d3d